### ISSUE TITLE
Configured versionable attributes are ignored -> everything is versioned

### ISSUE THREAD

**Issue by heikokrebs** [overtrue/laravel-versionable#88]
Hello @overtrue!

First of all, thank you very much for this great package!

With the latest release, an error has unfortunately crept into here: [src/Versionable.php Line 72](https://github.com/overtrue/laravel-versionable/commit/06f37ec3917fdf39fb5493a115c52caca8041dcd#diff-98f26c6f28d945173123295e2c88e7d6ac3daeb6d6d24386a5403e6660287a2dR72)

This leads to the fact that all changed attributes of the model (-> `getDirty()`) are always stored in $replacements and so a version is created with every change - even if only attributes that are not noted in $versionable have been changed.

I think this line can just be removed? Why was it introduced?

Thanks a lot for your help!

**Comment by overtrue** [comment]
let me check it

**Comment by overtrue** [comment]
did you mean the first version of a model?

**Comment by heikokrebs** [comment]
> did you mean the first version of a model?

No, everytime the model is saved you add every model-attribute that was changed via the `$replacements = array_merge($this->getDirty(), $replacements);` to the `$replacements` array.
I dont understand why this is necessary or what the idea was behind it?

The changed attributes you get via `$this->getDirty()` are not compared against the attributes that are decalared in the `$versionable` array on the model. So anytime ANY attribute on the model was changed, a new version will be created - and not only when one or more attributes out of the `$versionable` array are changed.

### Example

We have model `Car` with attributes `color` & `horsepower`. I only want to track changes of `color` in my version, so i define `protected $versionable = ['color']; 


```php
// Model
class Car extends Model
{
    use Versionable;
    
    protected $versionable = ['color'];
    
    …
}

// Change something
$car = Car::find(1);
$car->update(['horsepower' => 200]);

// -> a version will be created, although horsepower is not tracked!
// -> $this->shouldBeVersioning() will be false, because no versionable attribute was changed BUT ! empty($replacements) will be true.
```

Now if i update the `horsepower` for a model and save it to the database, `$this->getDirty()` will return the changes in `horsepower` and the if-check `if ($this->shouldBeVersioning() || ! empty($replacements)) {…}` will be true, because `$replacements` is not empty because it contains the change of `horsepower`. So a new (unwanted) version will be saved.

**Comment by overtrue** [comment]
> > did you mean the first version of a model?
> 
> No, everytime the model is saved you add every model-attribute that was changed via the `$replacements = array_merge($this->getDirty(), $replacements);` to the `$replacements` array. I dont understand why this is necessary or what the idea was behind it?
> 
> The changed attributes you get via `$this->getDirty()` are not compared against the attributes that are decalared in the `$versionable` array on the model. So anytime ANY attribute on the model was changed, a new version will be created - and not only when one or more attributes out of the `$versionable` array are changed.
> 
> ### Example
> We have model `Car` with attributes `color` & `horsepower`. I only want to track changes of `color` in my version, so i define `protected $versionable = ['color'];
> 
> ```
> // Model
> class Car extends Model
> {
>     use Versionable;
>     
>     protected $versionable = ['color'];
>     
>     …
> }
> 
> // Change something
> $car = Car::find(1);
> $car->update(['horsepower' => 200]);
> 
> // -> a version will be created, although horsepower is not tracked!
> // -> $this->shouldBeVersioning() will be false, because no versionable attribute was changed BUT ! empty($replacements) will be true.
> ```
> 
> Now if i update the `horsepower` for a model and save it to the database, `$this->getDirty()` will return the changes in `horsepower` and the if-check `if ($this->shouldBeVersioning() || ! empty($replacements)) {…}` will be true, because `$replacements` is not empty because it contains the change of `horsepower`. So a new (unwanted) version will be saved.

This change is actually to be compatible with the manually created version. Although I think this function seems to be of little use. 


https://github.com/overtrue/laravel-versionable/blob/5.x/tests/ManualVersionTest.php#L36-L38

**Comment by overtrue** [comment]
@heikokrebs I've released version 5.3.2 and ignored this feature for now. The main logic takes priority. It's expected that the function of manually creating versions will be removed in the next major version. Thank you for your feedback.

**Comment by heikokrebs** [comment]
@overtrue thanks for your very fast help!
